module-system/sov-state/src/prover_storage.rs

```diff
@@ -67,6 +74,11 @@ impl<S: MerkleProofSpec> Storage for ProverStorage<S> {
         val
     }
 
+    fn get_state_root(&self, _witness: &Self::Witness) -> anyhow::Result<[u8; 32]> {
+        self.get_root_hash(self.db.get_next_version() - 1)
```

Comment:
I would use [safe arithmetic](https://doc.rust-lang.org/std/intrinsics/fn.saturating_sub.html) here as a wrapped integer might become an attack vector.

Reply:
+1, as we already have separate issue to tacke it.

Reply:
Since we don't have specified safe-subtraction (that is part of the issue), I propose that we let the person in charge cover that case, I will link the issue here so that we can see it there: #373